tools/coco_visualize.py

```diff
@@ -115,23 +127,27 @@ def main():
             print(path)
 
         masks = []
+        masks_colors = []
 
         for ann in annotations[image_info['id']]:
             bbox = ann['bbox']
             p1 = int(bbox[0]), int(bbox[1])
             p2 = int(bbox[2] + bbox[0]), int(bbox[3] + bbox[1])
 
+            bgr_color = palette[cat_id_to_color_id[ann['category_id']]][::-1]
+
             try:
-                cv2.putText(image, categories[ann['category_id']]['name'], p1, 1, 2, (255, 0, 0), 2)
+                cv2.putText(image, categories[ann['category_id']]['name'], p1, 1, bbox[2] * 0.01, bgr_color, 2)
```

Comment:
Why does the font scale depends on a bbox width?

Reply:
If there is a crowded scene with a lot of small objects there is a mess of overlapping text instances so that nothing can be recognized.